pkg/virt-operator/install-strategy/create.go

```diff
@@ -751,38 +806,50 @@ func createOrUpdateClusterRole(cr *rbacv1.ClusterRole,
 		return err
 	}
 
+	injectOperatorMetadata(kv, &cr.ObjectMeta, imageTag, imageRegistry, id, true)
+
 	cr = cr.DeepCopy()
 	obj, exists, _ := stores.ClusterRoleCache.Get(cr)
-
-	if exists {
-		cachedCr = obj.(*rbacv1.ClusterRole)
-	}
-
-	injectOperatorMetadata(kv, &cr.ObjectMeta, imageTag, imageRegistry, id, true)
 	if !exists {
+		setLastAppliedConfiguration(cr)
 		// Create non existent
 		expectations.ClusterRole.RaiseExpectations(kvkey, 1, 0)
 		_, err := rbac.ClusterRoles().Create(cr)
 		if err != nil {
 			expectations.ClusterRole.LowerExpectations(kvkey, 1, 0)
 			return fmt.Errorf("unable to create clusterrole %+v: %v", cr, err)
 		}
+
 		log.Log.V(2).Infof("clusterrole %v created", cr.GetName())
-	} else if !objectMatchesVersion(&cachedCr.ObjectMeta, imageTag, imageRegistry, id, kv.GetGeneration()) {
+		return nil
+	}
+
+	cachedCr = obj.(*rbacv1.ClusterRole)
+	cachedCr.APIVersion = cr.APIVersion
+	cachedCr.Kind = cr.Kind
+
+	sync, obj, err := resourceShouldUpdate(cachedCr, cr)
+	if err != nil {
+		return err
+	}
+
+	if sync {
+		cr = obj.(*rbacv1.ClusterRole)
 		// Update existing, we don't need to patch for rbac rules.
-		_, err = rbac.ClusterRoles().Update(cr)
+		_, err := rbac.ClusterRoles().Update(cr)
 		if err != nil {
 			return fmt.Errorf("unable to update clusterrole %+v: %v", cr, err)
 		}
-		log.Log.V(2).Infof("clusterrole %v updated", cr.GetName())
 
+		log.Log.V(2).Infof("clusterrole %v updated", cr.GetName())
 	} else {
 		log.Log.V(4).Infof("clusterrole %v already exists", cr.GetName())
 	}
 
 	return nil
 }
 
+// TODO: should this be synced?
```

Comment:
Good question! Should configmaps be synced? I've waffled back and forth as I've thought about this. My perspective is that in a managed deployment, we probably should reconcile, but in clusters where KubeVirt is installed directly, I'm not so sure.

@rmohr care to weigh in here?

Reply:
ahh, this is my bad. I forgot to erase the comment. This function actually seems to not be needed, unless its for backwards compatibility we have a check so that this only syncs non ca configmaps. 

```
if cm.Name == components.KubeVirtCASecretName {
			continue
}
```

But the only configmap we create in the install strategy is the `kubevirt-ca` configmap. So maybe this should just be deleted?